aredis_om/model/model.py

```diff
@@ -563,7 +563,10 @@ def resolve_value(
                         separator_char,
                     )
                     return ""
-                if separator_char in value:
+                if isinstance(value, int):
+                    # This if will hit only if the field is a primary key of type int
+                    result = f"@{field_name}:[{value} {value}]"
```

Comment:
Interestingly (and painfully) enough, I've reach enough things recently to accept that we should probably start avoiding f-strings.

Let's not require it for this PR, but generally, we should remove them.